server.js

```diff
@@ -20,4 +35,89 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// Get all messages
+app.get("/messages", function (request, response) {
+  response.status(200).json(messages);
+});
+
+// Get all messages by search term
+app.get("/messages/search", function (request, response) {
+  if (request.query.text) {
+    let filteredMessages = messages.filter((msg) =>
+      msg.text.includes(request.query.text)
+    );
+    if (filteredMessages.length == 0) {
+      response.status(400).json({ msg: "No matching results" });
+    } else {
+      response.status(200).json(filteredMessages);
+    }
+  } else {
+    response.status(400).json({ msg: "Please enter search term" });
+  }
+});
```

Comment:
There's a lot of if's and elses going on in here. I'd try to refactor it to something a bit calmer. 

```
if (request.query.text) {
     return response.status(400).json({ msg: "Please enter search term" }); 
}
let filteredMessages = messages.filter((msg) =>
msg.text.includes(request.query.text)
 );
if (filteredMessages.length == 0) {
    return response.status(400).json({ msg: "No matching results" });
return response.status(200).json(filteredMessages);

```
A good rule of thumb to remember is that code is often written only once, but read many times, so readability it important.